support/ci/cargo_update.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+set -euo pipefail 
+ 
+# shellcheck source=./support/ci/shared.sh 
+source ./support/ci/shared.sh 
+
+branch="ci/cargo-update-$(date +"%Y%m%d")"
+git checkout -b "$branch"
+
+toolchain="$(get_toolchain)"
+
+install_rustup
+install_rust_toolchain "$toolchain"
+
+echo "--- :box: Cargo Update"
+cargo +"$toolchain" update
+echo "--- :box: Cargo Check"
+cargo +"$toolchain" check --quiet --all --tests
+
+git add Cargo.lock
+
+if [ "$BUILDKITE" != "true" ]; then 
```

Comment:
This variable won't be defined at all if you're not on buildkite, so the script will abort on when we try to access it. Since it's only ever `true` or undefined, I'd suggest
 ```suggestion
if "${BUILDKITE:-false}"; then 
```
We don't need to make use of the [`test`](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/test.html) command (for which `[…]` is syntactic sugar), since [`true`](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/true.html) and [`false`](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/false.html) are POSIX commands that do what we want directly.